Frontend/src/services/cluster-service.tsx

```diff
@@ -0,0 +1,106 @@
+'use client';
+import { AxiosResponse } from "axios";
+import {authApi} from "./axios";
+
+interface ApiResponse<T> {
+  data: T;
+}
+
+interface ApiErrorResponse {
+  message: string;
+}
+
+export async function addNewCluster(name: string, description: string, host: string, port: string, token: string | null): Promise<ApiResponse<string> | ApiErrorResponse> {
+  try {
+    const result: AxiosResponse<any> = await authApi({
+      method: "post",
+      url: "/backend/clusters",
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": "Bearer " + token,
+      },
+      data: {
+        name,
+        description,
+        host,
+        port,
+      },
+    });
+
+    return {
+      data: result.data,
+    };
+  } catch (err: any) {
+    if (err.response) {
+      return {
+        message: err.response.data.message,
+      };
+    } else {
+      return Promise.reject();
+    }
+  }
+}
+
+export async function getAllClusters(userID: string) {
+  try {
+    const result = await authApi({
+      method: "get",
+      url: `/backend/clusters/myClusters/${userID}`,
+    }).then((res) => res.data);
+    return {
+      data: result.data,
+    };
+  } catch (err) {
+    return Promise.reject();
+  }
+}
+
+export async function getCluster(clusterID: string) {
+  try {
+    const result = await authApi({
+      method: "get",
+      url: `/backend/clusters/${clusterID}`,
+    }).then((res) => res.data);
+    return {
+      data: result.data,
+    };
+  } catch (err) {
+    return Promise.reject();
```

Comment:
Shouldn't we log the error message here?

Reply:
Errors are handled in the calling function; this function simply returns the error.